subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c

```diff
@@ -1179,6 +1179,50 @@ uint32_t radio_bc_has_match(void)
 	return (NRF_RADIO->EVENTS_BCMATCH != 0);
 }
 
+#if defined(CONFIG_BT_CTLR_RNG_FAKE)
+static uint8_t rng_fake;
+
+static uint8_t rng_fake_xor(uint32_t value)
+{
+	uint8_t ms;
+
+	/* This is something that came of the guy writing it and who saw that
+	 * the sequence of returned value in the broadcaster sample seemed
+	 * random enough for him!
+	 */
+	ms = rng_fake >> 4;
+	rng_fake <<= 1;
+	rng_fake ^= (uint8_t)(value & 0xFF);
+	rng_fake ^= (uint8_t)((value >> 8) & 0xFF);
+	rng_fake ^= (uint8_t)((value >> 16) & 0xFF);
+	rng_fake ^= (uint8_t)((value >> 24) & 0xFF);
+	rng_fake ^= ~ms;
+
+	return rng_fake;
```

Comment:
There are a lot of well-documented (and analyzed) pseudo-random number generators. This look a bit like a LFSR. Can you help provide a bit more explanation of this generator?

Reply:
Explained in the comment already in code.